### PR TITLE
Fixed flaky CI tests by adding httpbin docker container to the github workflows

### LINKED ISSUE #327: [CCI][BUG]  Httpbin.org intermittently stops responding, which fails some tests
### What is the bug?
[httpbin.org](http://httpbin.org/) gives 504 error intermittently, and since [some tests](https://github.com/opensearch-project/opensearch-py/blob/409c883013c3cbae45cf12ecf501af2851ec53f7/test_opensearchpy/test_connection.py#LL871) in the opensearch-py check the connection to [httpbin.org](http://httpbin.org/), the whole ci fails because the tests cannot send a request to this third party service.

### How can one reproduce the bug?
Try to run the test several times

### What is the expected behavior?
Tests will throw:
`
test_opensearchpy/test_connection.py::TestConnectionHttpbin::test_urllib3_connection - opensearchpy.exceptions.TransportError: TransportError(504, '<html>\r\n<head><title>504 Gateway Time-out</title></head>\r\n<body>\r\n<center><h1>504 Gateway Time-out</h1></center>\r\n</body>\r\n</html>\r\n')
FAILED test_opensearchpy/test_async/test_connection.py::TestConnectionHttpbin::test_aiohttp_connection - opensearchpy.exceptions.ConnectionTimeout: ConnectionTimeout caused by - TimeoutError()
`

### What is your host/environment?
macOS Monterey 12.2.1 (M1 chip)
Python v3.11

### Do you have any screenshots?
image

### Do you have any additional context?
[httpbin.org 1](http://httpbin.org/) may be experiencing high load, which could be causing the server to respond slowly or not at all

### PR DESCRIPTION
**Pull request by saimedhi**
### Description
Fixed flaky CI tests by adding httpbin docker container to the github workflows

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes #327 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by dblock — COMMENTED** [review]
I think this works, but I don't love it that we now require everyone to run a web server to pass tests.

How about something like https://pythonbasics.org/webserver/? Start a 3-line python web server in the test setup?

File: DEVELOPER_GUIDE.md [hunk 1]
```diff
@@ -18,6 +18,11 @@ $ nox -rs format
 # Run the test suite
 $ nox -rs test
 ```
+Note that unit tests require docker to be installed and httpbin container running. 
+
+```
+docker run --name httpbin -d -p 8000:80 kennethreitz/httpbin
+```
```
[the remark this answers is not in the document]
Comment (saimedhi) [reply DEVELOPER_GUIDE.md:25]: @dblock, I'll look into the approach you've recommended, but I'd want to point out that CI tests in opensearch-py client requires opensearch container running in docker. I mean it already uses docker.

[the remark this answers is not in the document]
Comment (dblock) [reply DEVELOPER_GUIDE.md:25]: That's a very fair point, but those are integration tests running against multiple versions of the server and docker makes sense for something that takes a while to setup and run.

**Comment by dblock** [comment]
Having to click retry a bunch of times in another PR I am motivated to see this get finished :) LMK if you need help @saimedhi?

**Pull request closed without merging**